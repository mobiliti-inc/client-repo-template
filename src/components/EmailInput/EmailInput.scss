@import "~variables";

.input-validation-message-label{
   @include input-validation-text;
}