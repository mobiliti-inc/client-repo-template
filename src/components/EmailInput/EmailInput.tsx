import React from 'react';

import TextInput, { INPUT_TYPES } from '../TextInput/TextInput';
import styles from './EmailInput.scss';

class EmailInput extends React.Component {
	state = {
		value: '',
		error: '',
	};

	static getDerivedStateFromProps(props, state) {
		if (props.value && state.value !== props.value) {
			return {
				value: props.value,
			};
		}
		return null;
	}

	inputChange = (event) => {
		const { value } = event.target;
		const error = this.validateInput(value);
		this.setState({
			value,
			error,
		});
		this.props.onChange({ email: value, emailError: error });
	}

	validateInput = (value) => {
		let error;
		if (!value.length) {
			error = 'Please enter your email';
		} else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(value)) {
			error = 'Please enter a valid email';
		} else {
			error = '';
		}
		return error;
	}

	render() {
		const { value, error } = this.state;
		const { shouldShowCheckMark } = this.props;
		return (
			<div>
				<TextInput
					{...this.props}
					type={INPUT_TYPES.EMAIL}
					isValid={!error.length}
					onChange={this.inputChange}
					value={value}
					inputDidChange={shouldShowCheckMark}
				/>
				{ error.length > 0 &&
				<label styleName="input-validation-message-label">
					{error}
				</label>
				}
			</div>);
	}
}

EmailInput.defaultProps = {
	shouldShowCheckMark: false
};

EmailInput.propTypes = {
	onChange: PropTypes.func.isRequired,
	shouldShowCheckMark: PropTypes.bool
};
export default CSSModules(EmailInput, styles, { allowMultiple: true });
