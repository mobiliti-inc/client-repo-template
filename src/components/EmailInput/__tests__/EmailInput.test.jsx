import EmailInput from '../EmailInput';

describe('EmailInput', () => {
	const initialProps = {
		id: 'email',
		name: 'email'
	};
	it('renders', () => {
		const tree = shallow(
			<EmailInput
				onChange={jest.fn()}
				{...initialProps}
			/>
		);

		matchSnapshot(tree);
	});

	it('sets the correct value when inputChange method is called', () => {
		const subject = shallow(
			<EmailInput
				{...initialProps}
				onChange={jest.fn()}
			/>
		);

		const event = {
			target: {
				value: 'test@mobiliti.com'
			}
		};
		expect(subject.instance().state.value).toEqual('');
		subject.instance().inputChange(event);
		expect(subject.instance().state.value).toEqual('test@mobiliti.com');
	});

	it('renders with empty value and no error', () => {
		const tree = shallow(
			<EmailInput
				{...initialProps}
				onChange={jest.fn()}
			/>
		);

		expect(tree.instance().state.value).toEqual('');
		expect(tree.instance().state.error).toEqual('');
	});

	it('sets appropiate error message to Please enter your email when inputChange is called with an empty value', () => {
		const tree = shallow(
			<EmailInput
				{...initialProps}
				onChange={jest.fn()}
			/>
		);

		expect(tree.instance().state.error).toEqual('');
		tree.instance().inputChange({ target: { value: '' } });
		expect(tree.instance().state.error).toEqual('Please enter your email');
	});

	it('sets appropiate error message to be Please enter a valid email when inputChange is called with an empty value', () => {
		const tree = shallow(
			<EmailInput
				{...initialProps}
				onChange={jest.fn()}
			/>
		);

		expect(tree.instance().state.error).toEqual('');
		tree.instance().inputChange({ target: { value: 'aa #' } });
		expect(tree.instance().state.error).toEqual('Please enter a valid email');
	});
});
