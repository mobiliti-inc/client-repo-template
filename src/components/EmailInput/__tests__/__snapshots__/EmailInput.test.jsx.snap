// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput renders 1`] = `
<div>
  <TextInput
    changePasswordVisibilityStatus={[Function]}
    className=""
    dark={false}
    disabled={false}
    errorMessage=""
    id="email"
    inputDidChange={false}
    isValid={true}
    label=""
    name="email"
    onBlur={null}
    onChange={[Function]}
    placeholder=""
    shouldShowCheckMark={false}
    shouldShowVisibility={false}
    type="email"
    value=""
  />
</div>
`;
